src/index.ts

```diff
@@ -195,6 +198,17 @@ async function runUpgradeScripts(scripts: UpgradeScript[]) {
 async function runUpgradeScript(script: UpgradeScript) {
     try {
         await script.script();
+        if (isLocalDevelopment) {
+            // run upgrade scripts twice locally to ensure that the scripts are idempotent
+            const diffBefore = getGitDiff();
+            await script.script();
+            const diffAfter = getGitDiff();
+
+            if (diffBefore !== diffAfter) {
+                console.error(`❌ ${script.name}: Changes detected when executing script twice`);
+                process.exit(-1);
+            }
```

Comment:
Do we want / need this check? 

The problem is that some scripts rely on EsLint for cleanup. These scripts fail now, making it more complicated to implement scripts. E.g., [replace-gridcoldef-import.ts](https://github.com/vivid-planet/comet-upgrade/blob/3aed8956d4386da1989fca8ba4648f57d8488ccf/src/v7/replace-gridcoldef-import.ts#L36) just adds an import declaration in the end (no matter if it's already present), since ESLint will clean it up

Reply:
It's a nice idea, but I would still remove it, since upgrade scripts should be easy to write (otherwise devs won't do it). Also, it should be noticeable that a script doesn't by running it twice.

Reply:
If we want the git diff feature, we could move the second run to the single script run feature and diff after executing eslint

Reply:
I removed it in 785e3dfb0feb518fdd487f429854af82453387e6